infra/trick-cluster.yaml

```diff
@@ -0,0 +1,56 @@
+version: "3.8"
+
+services:
+  #### Base trick runtimes
+
+  cli-runtime:
+    scale: 0
+    image: nasatrick/cli-runtime
+    build:
+      context: ..
+      dockerfile: ${DOCKERFILE}
+      target: cli-runtime
+    ports:
+      - "9001:9001" # Trick variable server
+      - "8888:8888" # civetweb REST API + websocket server
+    volumes:
+      - ../trick_sims:/opt/trick_sims
+
+  gui-runtime:
+    scale: 0
+    image: nasatrick/gui-runtime
+    build:
+      context: ..
+      dockerfile: ${DOCKERFILE}
+      target: gui-runtime
+    volumes:
+      - /dev/shm:/dev/shm # not necessary, but used by vnc if provided
+    ports:
+      - "5901:5901" # VNC
+      - "6901:6901" # noVNC
+#      - "9001:9001" # Trick variable server
+#      - "8888:8888" # civetweb REST API + websocket server
+
+  #### Example distributed runtime for Trick Cannon web-sim
+  cannon-sim-cli-runtime:
+    scale: 1
+    image: nasatrick/cannon-sim-cli-runtime
+    extends:
+      service: cli-runtime
+    build:
+      target: cannon-sim-cli-runtime
+    networks:
+      sim:
+
+  cannon-sim-gui-runtime:
+    scale: 1
+    image: nasatrick/cannon-sim-gui-runtime
+    extends:
+      service: gui-runtime
+    build:
+      target: cannon-sim-gui-runtime
+    networks:
+      sim:
```

Comment:
should probably refactor this into the docker-compose and use profiles to launch just these svc.

this will avoid duplication of code between this docker-compose and the other one